x/interchainquery/keeper/keeper.go

```diff
@@ -45,69 +45,72 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	// Confirm the connectionId and chainId are valid
+	if connectionId == "" {
+		errMsg := "[ICQ Validation Check] Failed! connection id cannot be empty"
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
 	}
-	// height must be 0
-	if height != 0 {
-		k.Logger(ctx).Error("[ICQ Validation Check] Failed! height for interchainquery must be 0 (we exclusively query at the latest height on the host zone)")
+	if !strings.HasPrefix(connectionId, "connection") {
+		errMsg := "[ICQ Validation Check] Failed! connection id must begin with 'connection'"
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
 	}
-	// chain_id cannot be empty
-	if chain_id == "" {
-		k.Logger(ctx).Error("[ICQ Validation Check] Failed! chain_id cannot be empty")
+	if chainId == "" {
+		errMsg := "[ICQ Validation Check] Failed! chain_id cannot be empty"
+		k.Logger(ctx).Error(errMsg)
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
 	}
-	// ======================================================================================================================
-
-	key := GenerateQueryHash(connection_id, chain_id, query_type, request, module, height)
-	existingQuery, found := k.GetQuery(ctx, key)
-	if !found {
-		if module != "" {
-			if _, exists := k.callbacks[module]; !exists {
-				err := fmt.Errorf("no callback handler registered for module %s", module)
-				k.Logger(ctx).Error(err.Error())
-				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "no callback handler registered for module")
-			}
-			if exists := k.callbacks[module].Has(callback_id); !exists {
-				err := fmt.Errorf("no callback %s registered for module %s", callback_id, module)
-				k.Logger(ctx).Error(err.Error())
-				return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "no callback handler registered for module")
 
-			}
+	// Confirm the module and callbackId exist
+	if module != "" {
+		if _, exists := k.callbacks[module]; !exists {
+			err := fmt.Errorf("no callback handler registered for module %s", module)
+			k.Logger(ctx).Error(err.Error())
+			return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "no callback handler registered for module")
 		}
-		newQuery := k.NewQuery(ctx, module, connection_id, chain_id, query_type, request, period, callback_id, ttl, height)
-		k.SetQuery(ctx, *newQuery)
+		if exists := k.callbacks[module].Has(callbackId); !exists {
+			err := fmt.Errorf("no callback %s registered for module %s", callbackId, module)
+			k.Logger(ctx).Error(err.Error())
+			return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "no callback handler registered for module")
+		}
+	}
+
+	// Check to see if the query already exists
+	key := GenerateQueryHash(connectionId, chainId, queryType, request, module, height)
+	query, found := k.GetQuery(ctx, key)
 
+	// If this is a new query, build the query object
+	if !found {
```

Comment:
Could we remove this block entirely?

Reply:
Can you add a GH suggestion for what you're envisioning?

Reply:
Ah nevermind - the thought was to remove the if-else block of code but I realized the height is also different (not just the TTL)

Reply:
The height is actually the same each time (it's 0).

The idea here is we check if we already have a query in the store - and if we do, we just update the TTL. But if we don't, then we make a new query object and store that one.

If there's a way to format the code that would make that a little more obvious, please lmk!